utils/vptr/include/virtual_ptr.hpp

```diff
@@ -228,10 +230,15 @@ class PointerMapper {
   }
 
   /* get_buffer.
-   * Returns a buffer from the map using the pointer address
+   * Returns a pointer to a buffer from the map using the pointer address
    */
-  buffer_t get_buffer(const virtual_pointer_t ptr) {
-    return get_node(ptr)->second._b;
+  template <typename buffer_allocator = default_buffer_allocator>
+  cl::sycl::buffer<buffer_data_type, 1, buffer_allocator> get_buffer(
+      const virtual_pointer_t ptr) {
+    using buffer_t = cl::sycl::buffer<buffer_data_type, 1, buffer_allocator>;
+    auto buf = &get_node(ptr)->second._b;
+    buffer_t *b = static_cast<buffer_t *>(buf);
```

Comment:
I have changed the code to use the copy constructor.

I have also removed the template argument for the allocator type.
However, to use the `typed_allocator`, I need to pass it an allocator type, so I still can't use it without one:
`template <typename T, typename AllocatorT>
class typed_allocator : public base_allocator {}`

Maybe we can use the `base_allocator` instead?